### PR TITLE
[ANCHOR-374] Add assets config validations

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

- Add assets config validations
- Migrated asset config classes to assets packages

### Context

- Prepare [ANCHOR-794] - Add comments as document to `assets.yaml` and removed unused fields

### Testing

- `./gradlew test`

### Documentation

N/A.

### Known limitations

N/A

[ANCHOR-794]: https://stellarorg.atlassian.net/browse/ANCHOR-794?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/src/main/java/org/stellar/anchor/util/AssetValidator.java [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    }
+
+    // Validate fiat assets
+    for (FiatAssetInfo fiatAsset : assetService.getFiatAssets()) {
+      validateFiatAsset(assetService, fiatAsset);
+    }
+  }
+
+  // Check if deposit is enabled for the asset
+  public static boolean isDepositEnabled(DepositWithdrawInfo info) {
```
Comment (JiahuiWho) [inline core/src/main/java/org/stellar/anchor/util/AssetValidator.java:45]: We might want to put it under AssetHelper or AssetUtil?

File: platform/src/main/resources/config/anchor-asset-default-values.yaml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        # the maximum amount that the user can receive.
+        min_amount: 0
+        max_amount: 1000
+
+    # The configuration for the SEP-38 transactions.
+    sep38:
+      # `true` if the asset supports the SEP-38.
+      enabled: false
+      #
+      # country_codes:
```
Comment (JiahuiWho) [inline platform/src/main/resources/config/anchor-asset-default-values.yaml:107]: nit uncomment

Comment (lijamie98) [reply platform/src/main/resources/config/anchor-asset-default-values.yaml]: Removed.

File: core/src/main/java/org/stellar/anchor/util/AssetValidator.java (hunk 1 above)
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/util/AssetValidator.java]: Moved.

File: core/src/main/java/org/stellar/anchor/util/AssetValidator.java [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+  static void validateDepositWithdrawInfo(DepositWithdrawInfo dwInfo, String assetId)
+      throws InvalidConfigException {
+
+    // Validate withdraw fields
+    if (isWithdrawEnabled(dwInfo)) {
+      if (isEmpty(dwInfo.getWithdraw().getMethods())) {
+        throw new InvalidConfigException(
+            format("No withdraw methods defined for asset %s", assetId));
+      }
+      // Check for duplicate SEP-6 withdrawal types
```
Comment (JiahuiWho) [inline core/src/main/java/org/stellar/anchor/util/AssetValidator.java:171]: nit remove mention of "SEP-6"

File: platform/src/main/resources/config/anchor-asset-default-values.yaml [hunk 4]
```diff
@@ -103,9 +103,6 @@ items:
     sep38:
       # `true` if the asset supports the SEP-38.
       enabled: false
-      #
-      # country_codes:
-
```
Comment (JiahuiWho) [inline platform/src/main/resources/config/anchor-asset-default-values.yaml]: Oh Stellar Asset doesn't have country_codes?

**Review by JiahuiWho — APPROVED** [review]

File: core/src/main/java/org/stellar/anchor/util/AssetValidator.java (hunk 3 above)
Comment (lijamie98) [reply core/src/main/java/org/stellar/anchor/util/AssetValidator.java:171]: Removed.

File: platform/src/main/resources/config/anchor-asset-default-values.yaml (hunk 4 above)
Comment (lijamie98) [reply platform/src/main/resources/config/anchor-asset-default-values.yaml]: Removed.

**Pull request merged**